### ISSUE TITLE
Preprocess the Data

### ISSUE THREAD

**Issue by ahmetrnn** [ahmetrnn/MachineLearningProject#2]
Your task is to prepare the dataset for machine learning by cleaning, transforming, and splitting the data. This includes handling missing values, encoding categorical variables, and creating training/testing sets.

Handle Missing Values (if any)
• Goal: Deal with missing data to avoid errors during model training.
• How:
• First, re-check missing values using: df.isnull().sum()
• Then apply an imputation strategy:
• Use .fillna() or SimpleImputer from sklearn.impute
• Choose strategy based on column type:
• Numerical → mean or median
• Categorical → most frequent (mode)

• Example tools:
• from sklearn.impute import SimpleImputer
• Or simply df['col'].fillna(df['col'].mean(), inplace=True)

Encode Categorical Variables
• Goal: Convert non-numeric (object) features into numeric format.
• How:
• First, identify categorical columns: df.select_dtypes(include='object').columns
• Use:
• pd.get_dummies() for One-Hot Encoding (best for non-ordinal categories like “Department”)
• LabelEncoder for Label Encoding if the variable has ordinal meaning
• Save encoded data to a new DataFrame or overwrite old columns

• Example tools:
• from sklearn.preprocessing import LabelEncoder
• pd.get_dummies(df, columns=[...])

Split the Data into Train and Test Sets
• Goal: Prepare separate sets for training and evaluation
• How:
• Define X (features) and y (target column: Attrition)
• Use train_test_split() from sklearn.model_selection
• Typical split ratio: test_size=0.2 (20% test), random_state=42 for reproducibility

• Example tools:
• from sklearn.model_selection import train_test_split
• X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

Notes
• Before splitting, ensure the final dataset has:
• No missing values
• Only numeric features
• Clear separation between input features (X) and target (y)
• Make sure the target column (Attrition) is transformed into 0 and 1 if it’s still string (“Yes”/“No”)
